ChaosMod/Effects/db/Screen/ScreenRealFirstPerson.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+    Effect by Chanka
+*/
+
+#include <stdafx.h>
+
+static Cam eCamera;
+static Entity boneProxy;
+static int state = 0;
+
+static bool SafetyCheck()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	//Removes camera on player switch, death, and potential non-existence
+	if (IS_PLAYER_SWITCH_IN_PROGRESS() || IS_PLAYER_DEAD(PLAYER_ID()) || !ENTITY::DOES_ENTITY_EXIST(playerPed)
+	    || !ENTITY::IS_ENTITY_VISIBLE(playerPed))
+	{
+		return true;
+	}
+	return false;
+}
+
+//OnStart and OnStop code moved to separate functions at Rylxnd's request
+static void RemoveCamera()
+{
+	RENDER_SCRIPT_CAMS(false, false, 0, true, false, 0);
+	CAN_SET_EXIT_STATE_FOR_CAMERA(true);
+	if (DOES_ENTITY_EXIST(boneProxy))
+	{
+		DELETE_ENTITY(&boneProxy);
+	}
+	if (DOES_CAM_EXIST(eCamera))
+	{
+		SET_CAM_ACTIVE(eCamera, false);
+		DESTROY_CAM(eCamera, true);
+	}
+}
+
+static void SetupCamera()
+{
+	// Not sure of a direct way to obtain bone rotation. Attaching a prop is a workaround.
```

Comment:
`ENTITY::GET_ENTITY_BONE_ROTATION(..., PED::GET_PED_BONE_INDEX(...))` 👍

Reply:
Implemented it.